### PR TITLE
Document message naming, including swift_prefix

### PR DESCRIPTION
**Pull request by tbkka**
Address #562

### CODE REVIEW

**Comment by tbkka** [comment]
@Arial7:  Do you think this suffices?

File: Documentation/API.md [hunk 1]
```diff
@@ -93,10 +93,46 @@ public struct Example: SwiftProtobuf.Message {
...
+in your proto file:
+```protobuf
+   syntax = "proto3";
+   package my_company.cool_project;
+   option swift_prefix="My";
+   message VeryImportant {
+      ...
+   }
+```
+will generate a struct named `MyFooBar`.
```
Comment (allevato) [inline Documentation/API.md]: `MyVeryImportant`?

File: Documentation/API.md [hunk 2]
```diff
@@ -125,7 +161,10 @@ want to override the default generated behavior for any reason:
 Proto enums are translated to Swift enums in a fairly straightforward manner.
 The resulting Swift enums conform to the `SwiftProtobuf.Enum` protocol which extends
 `RawRepresentable` with a `RawValue` of `Int`.
-The name of the Swift enum is copied directly from the name in the proto file.
+
+The name of the Swift enum is copied directly from the name in the proto file,
+prefixed with the package name or the name from `option swift_prefix`
+as documented above for messages.
 If that name would cause problems for the generated code, the word `Enum` will
 be appended to the name.
```
Comment (allevato) [inline Documentation/API.md]: We should probably also mention case stemming here and give an example.

File: Documentation/API.md (hunk 1 above)
Comment (tbkka) [inline Documentation/API.md]: Fixed (changed the code example to match the docs and the previous example).

File: Documentation/API.md (hunk 2 above)
Comment (tbkka) [inline Documentation/API.md]: I've gone through and made a pass at this.  Feedback appreciated.

File: Documentation/API.md [hunk 3]
```diff
@@ -214,6 +199,61 @@ public enum MyEnum: SwiftProtobuf.Enum {
...
+```swift
+enum TestEnum {
+  case foo = 0
+  case bar_1 = 1
+  case baz = 2
+  case bar_n3 = -3 // 'n' for "negative"
+}
+```
+
+Note #1: As explained in the protobuf language guide,
```
Comment (thomasvl) [inline Documentation/API.md]: My $0.02, i'd liken't leave tis off, as it isn't really about our generated API, and we probably don't want to have to duplicate all the google/protobuf docs about good/bad proto design.

Comment (tbkka) [inline Documentation/API.md]: Removed.

**Pull request merged**